sunpy/coordinates/sun.py

```diff
@@ -100,13 +100,35 @@ def carrington_rotation_time(crot):
 
     Parameters
     ----------
-    crot : float
-        Carrington rotation number. Can be a fractional cycle number.
+    crot : int, float, numpy.array
+        Carrington rotation number. Can be a integral or fractional cycle number
+        or a Numpy array of such Carrington rotation numbers. If a single
+        Carrington rotation number is passed with multiple logitudes then
+        rotation times at each of those longitudes is calculated.
+
+    longitude : astropy.units.Quantity
+        Carrington longitude (in degrees).If a single longitude is passed with
+        multiple Carrington rotation numbers then the same rotation is applied
+        to each Carrington rotation number while calculating rotation times.
+        When this argument is passed Carrington rotation number(s) should be
+        strictly integral.
 
     Returns
     -------
     astropy.time.Time
     """
+    if longitude is not None:
+        crot = np.array(crot)
+        assert np.issubdtype(crot.dtype, int), "Carrington Number(s) should be of type int"
+        longitude = np.array(longitude.to('deg').value)
+        if crot.shape != longitude.shape:
+            if not longitude.shape or longitude.shape[0] == 1:
+                longitude = np.full_like(crot, longitude)
+            elif not crot.shape or crot.shape[0] == 1:
+                crot = np.full_like(longitude, crot)
+            else:
+                raise Exception("Array of rotation number and longitudes are of unequal length")
+        crot = crot + (1 - longitude/360)
```

Comment:
```suggestion
        crot = crot + (1 - longitude/(360*u.deg))
```

Reply:
This change raises an error since `Quantity` (even dimensionless ones) cannot be subtracted from `int` objects, `1` in this case. ( Even if the above line which converts it into `longitude` into a numpy array is removed ). I think we should rather avoid this since `crot` would be passed ahead as `Quantity` rather than a numpy array even if we fix this.

Reply:
You should switch to using `Quantity` everywhere